### PR TITLE
build: Remove deprecated flex driver and update the min K8s version to v1.16

### LINKED ISSUE #8076: Remove deprecated FlexVolume driver
We have been using CSI as the default driver for quite a while now, and it is now at feature parity with the FlexVolume or better. We can now stop supporting the old FlexVolume driver.

@Madhu-1 are you able to confirm CSI is at parity with FlexVolume?

### PR DESCRIPTION
**Pull request by travisn**
**Description of your changes:**
- The flex driver has been fully deprecated and thus removed from Rook. Before upgrading to v1.8, users will need to convert existing flex volumes from flex to csi volumes.
- In Rook v1.8 the min version of K8s supported is updated to 1.16. Users running on older versions of K8s are recommended to update to 1.16 or newer before updating to Rook v1.8.

Despite how large the PR is, the risk of backport conflicts doesn't seem too bad at this point.

**Which issue is resolved by this Pull Request:**
Resolves #8076 #8084

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/latest/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/latest/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

### CODE REVIEW

File: PendingReleaseNotes.md [hunk 1]
```diff
@@ -8,14 +8,9 @@ v1.8...
 
 ## Breaking Changes
 
-### Ceph
+- Flex driver is fully deprecated. If you are still using flex volumes, before upgrading to v1.8
+  you will need to convert them to csi volumes. See the flex conversion tool.
```
Comment (subhamkrai) [inline PendingReleaseNotes.md:12]: suggestion: add a link to the tool.

File: pkg/apis/ceph.rook.io/v1/annotations.go [hunk 2]
```diff
@@ -17,47 +17,78 @@ limitations under the License.
...
+	for k, v := range a {
+		if _, ok := t.Annotations[k]; !ok {
+			t.Annotations[k] = v
+		}
+	}
+}
+
+// Merge returns an Annotations which results from merging the attributes of the
+// original Annotations with the attributes of the supplied one. The supplied
+// Placement's attributes will override the original ones if defined.
```
Comment (subhamkrai) [inline pkg/apis/ceph.rook.io/v1/annotations.go:82]:
```suggestion
// Annotations attributes will override the original ones if defined.
```
?

**Review by subhamkrai — DISMISSED** [review]
small nits

**Review by leseb — APPROVED** [review]
Hooray! Just one comment, otherwise LGTM!

File: PendingReleaseNotes.md (hunk 1 above)
Comment (travisn) [reply PendingReleaseNotes.md:12]: In the final release notes we can revise this and add a link. This is really just a placeholder for now.

**Pull request merged**